tests/mcp/video.spec.ts

```diff
@@ -79,34 +60,52 @@ for (const mode of ['isolated', 'persistent']) {
       ],
     });
 
-    expect(await client.callTool({
-      name: 'browser_navigate',
-      arguments: { url: server.HELLO_WORLD },
-    })).toHaveResponse({
-      code: expect.stringContaining(`page.goto('http://localhost`),
-    });
+    await navigateToTestPage(client, server);
+    await expect(async () => {
+      await produceFrames(client);
+      await checkIntermediateVideoFileExists(videosDir);
+    }).toPass();
+    await closeBrowser(client);
+  });
+}
 
-    await client.callTool({
-      name: 'browser_evaluate',
-      arguments: {
-        function: `async () => {
-          document.body.style.backgroundColor = "red";
-          await new Promise(resolve => setTimeout(resolve, 100));
-          document.body.style.backgroundColor = "green";
-          await new Promise(resolve => setTimeout(resolve, 100));
-          document.body.style.backgroundColor = "blue";
-          await new Promise(resolve => setTimeout(resolve, 100));
-        }`,
-      },
-    });
+async function navigateToTestPage(client: Client, server: any) {
+  expect(await client.callTool({
+    name: 'browser_navigate',
+    arguments: { url: server.HELLO_WORLD },
+  })).toHaveResponse({
+    code: expect.stringContaining(`page.goto('http://localhost`),
+  });
+}
 
-    expect(await client.callTool({
-      name: 'browser_close',
-    })).toHaveResponse({
-      code: expect.stringContaining(`page.close()`),
-    });
+async function closeBrowser(client: Client) {
+  expect(await client.callTool({
+    name: 'browser_close',
+  })).toHaveResponse({
+    code: expect.stringContaining(`page.close()`),
+  });
+}
 
-    const [file] = await fs.promises.readdir(videosDir);
-    expect(file).toMatch(/.*.\webm/);
+async function produceFrames(client: Client) {
+  expect(await client.callTool({
+    name: 'browser_evaluate',
+    arguments: {
+      function: `async () => {
+        document.body.style.backgroundColor = "red";
+        await new Promise(resolve => setTimeout(resolve, 100));
+        document.body.style.backgroundColor = "green";
+        await new Promise(resolve => setTimeout(resolve, 100));
+        document.body.style.backgroundColor = "blue";
+        await new Promise(resolve => setTimeout(resolve, 100));
+        return 'ok';
+      }`,
+    },
+  })).toHaveResponse({
+    result: '"ok"',
   });
 }
+
+async function checkIntermediateVideoFileExists(videosDir?: string) {
+  const files = await fs.promises.readdir(videosDir ?? test.info().outputPath('tmp', 'playwright-mcp-output'));
```

Comment:
Make a fixture for `test.info().outputPath('tmp', 'playwright-mcp-output')` path, otherwise it's hard to track where it comes from.